src/libmain/shared.cc

```diff
@@ -427,7 +427,7 @@ RunPager::RunPager()
     });
 
     pid.setKillSignal(SIGINT);
-
+    stdout = dup(STDOUT_FILENO);
```

Comment:
I think this will cause us to leak a file descriptor if we don't mark it as CLOEXEC, see https://github.com/NixOS/nix/pull/5683. On Linux, `dup3` can be used to atomically create this FD as CLOEXEC.

Reply:
I think the better option is the `fcntl` one I have pushed. Not sure it works on darwin or any other platform though, so maybe I need to hind it behind `#ifdef __linux__`